lib/internal/console/constructor.js

```diff
@@ -407,17 +406,11 @@ const consoleMethods = {
   },
 
   timeEnd(label = 'default') {
-    if (this[kInternalTimeLogImpl] === undefined)
-      this[kInternalTimeLogImpl] = FunctionPrototypeBind(timeLogImpl, this);
-
-    timeEnd(this._times, kTraceConsoleCategory, 'console.timeEnd()', kNone, this[kInternalTimeLogImpl], label, `time::${label}`);
+    timeEnd(this._times, kTraceConsoleCategory, 'console.timeEnd()', kNone, (label, formatted, args) => timeLogImpl(this, label, formatted, args), label, `time::${label}`);
   },
 
   timeLog(label = 'default', ...data) {
-    if (this[kInternalTimeLogImpl] === undefined)
-      this[kInternalTimeLogImpl] = FunctionPrototypeBind(timeLogImpl, this);
-
-    timeLog(this._times, kTraceConsoleCategory, 'console.timeLog()', kNone, this[kInternalTimeLogImpl], label, `time::${label}`, data);
+    timeLog(this._times, kTraceConsoleCategory, 'console.timeLog()', kNone, (label, formatted, args) => timeLogImpl(this, label, formatted, args), label, `time::${label}`, data);
```

Comment:
I believe it would be possible to add the timeLogImpl to the instance during construction time.
That way it is not required to recreate the function on each call.

Reply:
I tried adding inside the constructor but I always get `undefined`.

```diff
diff --git a/lib/internal/console/constructor.js b/lib/internal/console/constructor.js
index 6c34a9eded..8d88d429b0 100644
--- a/lib/internal/console/constructor.js
+++ b/lib/internal/console/constructor.js
@@ -90,6 +90,7 @@ const kBindStreamsEager = Symbol('kBindStreamsEager');
 const kBindStreamsLazy = Symbol('kBindStreamsLazy');
 const kUseStdout = Symbol('kUseStdout');
 const kUseStderr = Symbol('kUseStderr');
+const kInternalTimeLogImpl = Symbol('kInternalTimeLogImpl');
 
 const optionsMap = new SafeWeakMap();
 function Console(options /* or: stdout, stderr, ignoreErrors = true */) {
@@ -155,6 +156,8 @@ function Console(options /* or: stdout, stderr, ignoreErrors = true */) {
     });
   });
 
+  this[kInternalTimeLogImpl] = FunctionPrototypeBind(timeLogImpl, this);
+
   this[kBindStreamsEager](stdout, stderr);
   this[kBindProperties](ignoreErrors, colorMode, groupIndentation);
 }
@@ -371,11 +374,11 @@ function createWriteErrorHandler(instance, streamSymbol) {
   };
 }
 
-function timeLogImpl(consoleRef, label, formatted, args) {
+function timeLogImpl(label, formatted, args) {
   if (args === undefined) {
-    consoleRef.log('%s: %s', label, formatted);
+    this.log('%s: %s', label, formatted);
   } else {
-    consoleRef.log('%s: %s', label, formatted, ...new SafeArrayIterator(args));
+    this.log('%s: %s', label, formatted, ...new SafeArrayIterator(args));
   }
 }
 
@@ -406,11 +409,11 @@ const consoleMethods = {
   },
 
   timeEnd(label = 'default') {
-    timeEnd(this._times, kTraceConsoleCategory, 'console.timeEnd()', kNone, (label, formatted, args) => timeLogImpl(this, label, formatted, args), label, `time::${label}`);
+    timeEnd(this._times, kTraceConsoleCategory, 'console.timeEnd()', kNone, this[kInternalTimeLogImpl], label, `time::${label}`);
   },
 
   timeLog(label = 'default', ...data) {
-    timeLog(this._times, kTraceConsoleCategory, 'console.timeLog()', kNone, (label, formatted, args) => timeLogImpl(this, label, formatted, args), label, `time::${label}`, data);
+    timeLog(this._times, kTraceConsoleCategory, 'console.timeLog()', kNone, this[kInternalTimeLogImpl], label, `time::${label}`, data);
   },
 
   trace: function trace(...args) {
```

```
node:internal/util/debuglog:206
    logImp(label, formatted, args);
    ^

TypeError <Object <Object <[Object: null prototype] {}>>>: logImp is not a function
    at timeLogImpl (node:internal/util/debuglog:206:5)
    at timeLog (node:internal/util/debuglog:313:5)
    at console.timeLog (node:internal/console/constructor:423:5)
    at Object.<anonymous> (/home/h4ad/Projects/opensource/node-copy-4/test/parallel/test-console-with-frozen-intrinsics.js:18:9)
```